dexbot/strategies/relative_orders.py

```diff
@@ -118,57 +167,106 @@ def update_orders(self):
+            return
+
         orders = self.fetch_orders()
 
+        # Detect complete fill, order expiration, manual cancel, or just init
+        need_update = False
         if not orders:
-            self.update_orders()
+            need_update = True
         else:
-            orders_changed = False
-
             # Loop trough the orders and look for changes
             for order_id, order in orders.items():
                 current_order = self.get_order(order_id)
 
                 if not current_order:
-                    orders_changed = True
-                    self.write_order_log(self.worker_name, order)
+                    need_update = True
+                    self.log.debug('Could not found order on the market, it was filled, expired or cancelled')
+                    # FIXME: writing a log entry is disabled because we cannot distinguish an expired order
+                    #        from filled
+                    #self.write_order_log(self.worker_name, order)
+                elif self.is_reset_on_partial_fill:
+                    # Detect partially filled orders;
+                    # on fresh order 'for_sale' is always equal to ['base']['amount']
+                    if current_order['for_sale']['amount'] != current_order['base']['amount']:
+                        diff_abs = current_order['base']['amount'] - current_order['for_sale']['amount']
+                        diff_rel = diff_abs / current_order['base']['amount']
+                        if diff_rel >= self.partial_fill_threshold:
+                            need_update = True
+                            self.log.info('Partially filled order detected, filled {:.2%}'.format(diff_rel))
+                            # FIXME: need to write trade operation; possible race condition may occur: while
+                            #        we're updating order it may be filled futher so trade log entry will not
+                            #        be correct
+
+        if (self.is_reset_on_market_trade and
+            isinstance(event, FilledOrder)):
+            self.log.debug('Market trade detected, updating orders')
+            need_update = True
+
+        if self.is_reset_on_price_change:
+            center_price = self.calculate_center_price(
+                None,
+                self.is_asset_offset,
+                self.spread,
+                self['order_ids'],
+                self.manual_offset
+            )
+            diff = (self.center_price - center_price) / self.center_price
+            diff = abs(diff)
+            if diff >= self.price_change_threshold:
+                self.log.debug('Center price changed, updating orders. Diff: {:.2%}'.format(diff))
+                need_update = True
 
-            if orders_changed:
-                self.update_orders()
-            else:
-                self.log.info("Orders correct on market")
+        if need_update:
+            self.update_orders()
+        else:
+            pass
```

Comment:
The point of logging "Orders correct on market" is that it's used as a status for the worker in the GUI. 
In a case where the strategy is initialized but no action is required to be done after it, a status message "Initializing Relative Orders" will get stuck on the GUI and that will just confuse the user. 

I implemented a fix for this here https://github.com/Codaone/DEXBot/commit/347eedf4d3443d1a86c84cc9ffac89f10375e3e2#diff-472e57ad5bbb88571e326e56670648b7R173 but I think it's going to take a while until I can merge that to master, so could just copy the change from there?